File: ITing-frontend/src/services/notificationPreferenceService.js
import axiosInstance from "../utils/axiosInstance";

const notificationPreferenceService = {
  /** GET /api/candidate/notification-preferences */
  get: async () => {
    const res = await axiosInstance.get("/candidate/notification-preferences");
    return res?.data ?? res;
  },

  /** PUT /api/candidate/notification-preferences (partial update) */
  update: async (payload) => {
    const res = await axiosInstance.put(
      "/candidate/notification-preferences",
      payload
    );
    return res?.data ?? res;
  },
};

export default notificationPreferenceService;

File: ITing-backend/src/main/java/com/iting/jobportal/notification/service/NotificationPreferenceService.java
package com.iting.jobportal.notification.service;

import com.iting.jobportal.notification.dto.NotificationPreferenceDto;

public interface NotificationPreferenceService {

    /** Lấy preferences của user; nếu chưa có row → tạo default rồi trả về. */
    NotificationPreferenceDto getOrCreate(Long accountId);

    /** Update preferences. Field null trong DTO → giữ giá trị cũ (partial update). */
    NotificationPreferenceDto update(Long accountId, NotificationPreferenceDto dto);
}
